### PR TITLE
Multiple functionalities integration and bug fixes

### PR DESCRIPTION
**Pull request by nicolasp9-dev**

### CODE REVIEW

**Comment by nicolasp9-dev** [comment]
Home screen, new profile version, request for objects in db (geopoint), favors details view, complete espresso test of the app, first version of the map

**Comment by Jmion** [comment]
Hi Nicola,

Could you please detail a little bit more what has changed. I would be very happy to review this, but with 60 files changed and over 2800 lines of code added 800 removed I think that having a little bit of context on what was done would be good and save us (reviewer a lot of time).


Great job with Code coverage 👍 💯 🥇 Amazing work!

**Comment by nicolasp9-dev** [comment]
Hi Jeremy,
I’ll do a small summary of what I have done, that’s a great idea :
- creation of home screen, which is now the landing page after you logged in, where you can see lasts favors, more functionality can be add later (recent messages...)
- intégration of the list of favors as a menu element to see the favors you posted (My favors)
- creation of a parent to the favors list for map functionality switch integration
- creation of a complete espresso test of the app which check a lot of things, it’ll need to be adapted when we add some new functionalities
- new kind of request can be done, for example a single user
- a more adapted observable array was created to not have to override 5 methods when we want to observe an array
- DatabaseEntity are observable
- Charline functionality is included and test
- Map is loading but without displaying points

**Comment by Jmion** [comment]
Thanks for the details. I will look at the PR. Might take me a little bit of time, since it is quite a large PR.

File: app/src/main/java/ch/epfl/sweng/favors/favors/FavorsFragment.java [hunk 1]
```diff
@@ -1,108 +1,68 @@
...
-    public void onNothingSelected(AdapterView<?> parent) {}
-
-    private void updateList(ObservableList<Favor> list){
-        listAdapter = new FavorListAdapter(this.getActivity(), (ObservableArrayList)list);
-        binding.favorsList.setAdapter(listAdapter);
-        listAdapter.notifyDataSetChanged();
+        }
     }
 
-    ObservableList.OnListChangedCallback listCallBack = new ObservableList.OnListChangedCallback<ObservableList<Favor>>() {
```
Comment (Jmion) [inline app/src/main/java/ch/epfl/sweng/favors/favors/FavorsFragment.java]: 👍 Great

File: app/src/main/java/ch/epfl/sweng/favors/favors/FavorsList.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
+package ch.epfl.sweng.favors.favors;
```
Comment (Jmion) [inline app/src/main/java/ch/epfl/sweng/favors/favors/FavorsList.java]: OK. Clever design

File: app/src/main/java/ch/epfl/sweng/favors/favors/MyFavorsFragment.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import ch.epfl.sweng.favors.authentication.Authentication;
+import ch.epfl.sweng.favors.database.Favor;
+import ch.epfl.sweng.favors.database.FavorRequest;
+import ch.epfl.sweng.favors.database.ObservableArrayList;
+import ch.epfl.sweng.favors.databinding.MyFavorsBinding;
+
+/**
+ * Fragment that displays the list of favor and allows User to sort it and to search in it
+ */
+public class MyFavorsFragment extends android.support.v4.app.Fragment {
```
Comment (Jmion) [inline app/src/main/java/ch/epfl/sweng/favors/favors/MyFavorsFragment.java]: Very usefull for the next step of the tokens

**Comment by Jmion** [comment]
# Content of this branch :
#292 #294 
Issues: 
#18 #285 #286 #287 #288 #289 #282 #283 #264 #263 #262

**Pull request merged**